### ISSUE TITLE
Tables using ao_row / ao_colum TABLEAM extraneously claim to have `cmin`, `cmax`, `xmin`, and `xmax`

### ISSUE THREAD

**Issue by d** [greenplum-db/gpdb#11844]
### Greenplum version or build
master (7.0 ALPHA as of writing, commit f724542ff98ae58a)

### TL;DR
AO tables do not contain system columns `cmin`, `cmax`, `xmin`, `xmax`, and we historically do not support projecting them (the columns are absent from `pg_attribute`). It seems like an oversight while merging with upstream PostgreSQL 12 that we've presented the tables with these 4 extra columns?

A query like `SELECT xmin FROM abuela` used to error out during parse analysis, but now it only errors out at runtime, when the table AM cannot perform the projection. This is borderline unacceptable, given that we now have a "feature" that _only works on empty tables_.

### Step to reproduce the behavior

```sql
CREATE TABLE abuela (a int) USING ao_row;
INSERT INTO abuela VALUES (0);
SELECT attrelid::regclass, attname, attnum FROM pg_attribute WHERE attrelid = 'abuela'::regclass;

```

### Expected behavior

```
 attrelid |    attname    | attnum 
----------+---------------+--------
 abuela   | gp_segment_id |     -7
 abuela   | tableoid      |     -6
 abuela   | ctid          |     -1
 abuela   | a             |      1
(4 rows)
```

### Actual behavior

```
 attrelid |    attname    | attnum 
----------+---------------+--------
 abuela   | gp_segment_id |     -7
 abuela   | tableoid      |     -6
 abuela   | cmax          |     -5
 abuela   | xmax          |     -4
 abuela   | cmin          |     -3
 abuela   | xmin          |     -2
 abuela   | ctid          |     -1
 abuela   | a             |      1
(8 rows)
```

**Comment by dpandhi-git** [comment]
tested in 5x,6x,7x for `the columns are absent from pg_attribute`. All versions have column `cmin, cmax, xmin, xmax` stored in pg_attribute

For 6x ,7x query does fail at runtime not during parse analyze 

for 7x we can look for improvement of not storing these additional columns `cmin, cmax, xmin, xmax` in pg_attribute for AO/AOCO tables

**Comment by Chibin** [comment]
I did the following on 6X (Greenplum Database 6.24.4+dev.12.g681ab6e5f1):

```sql
CREATE TABLE abuela (a int) with(appendonly=true) ;
INSERT INTO abuela VALUES (0);
SELECT attrelid::regclass, attname, attnum FROM pg_attribute WHERE attrelid = 'abuela'::regclass;
```

I get the following result:
```sql
tmarbin=# SELECT attrelid::regclass, attname, attnum FROM pg_attribute WHERE attrelid = 'abuela'::regclass;
 attrelid |    attname    | attnum
----------+---------------+--------
 abuela   | a             |      1
 abuela   | ctid          |     -1
 abuela   | xmin          |     -3
 abuela   | cmin          |     -4
 abuela   | xmax          |     -5
 abuela   | cmax          |     -6
 abuela   | tableoid      |     -7
 abuela   | gp_segment_id |     -8
(8 rows)

tmarbin=# select version();
                                                                                                      version
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 PostgreSQL 9.4.26 (Greenplum Database 6.24.4+dev.12.g681ab6e5f1 build dev) on x86_64-apple-darwin22.5.0, compiled by Homebrew clang version 16.0.5, 64-bit compiled on Jun 21 2023 11:05:41 (with assert checking)
(1 row)
```

**Comment by Chibin** [comment]
With the new changes I'll be proposing soon, the new behavior will be:

```sql
drop table if exists abuela;
NOTICE:  table "abuela" does not exist, skipping
CREATE TABLE abuela (a int) USING ao_row;
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 'a' as the Greenplum Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
-- verify that cmax, xmax, cmin, and xmin do not exist
SELECT attrelid::regclass, attname, attnum FROM pg_attribute WHERE attrelid = 'abuela'::regclass;
 attrelid |    attname    | attnum
----------+---------------+--------
 abuela   | gp_segment_id |     -7
 abuela   | tableoid      |     -6
 abuela   | ctid          |     -1
 abuela   | a             |      1
(4 rows)

INSERT INTO abuela VALUES (0);
-- show parser fails
SELECT count(xmin) FROM abuela;
ERROR:  column "xmin" does not exist
LINE 1: SELECT count(xmin) FROM abuela;
```
